magicbot/magicrobot.py

```diff
@@ -378,7 +383,9 @@ def test(self):
 
             self._update_feedback()
             self.robotPeriodic()
-            wpilib.Timer.delay(self.control_loop_wait_time)
+            delay.wait()
+
+        delay.free()
```

Comment:
Come to think of it, this should be in a `finally` block.

Reply:
Good point, though I feel like a finalizer is the way to go here instead.

Reply:
Actually, if you used a `with` context, that would be even more pythonic.

Reply:
Good point. I didn't feel like touching all the lines here, but it's arguably nicer.